I'm going to revert back to the earlier version of this pull request, since the setting is needed unconditionally in order to build:

```
/app/webkit/Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:57:40: error: ‘DEFAULT_VALUE_FOR_CanvasUsesAcceleratedDrawing’ was not declared in this scope
   57 | #define FEATURE_DEFAULT(featureName) ((DEFAULT_VALUE_FOR_ ## featureName) ? TRUE : FALSE)
      |                                        ^~~~~~~~~~~~~~~~~~
/app/webkit/Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp:1329:13: note: in expansion of macro ‘FEATURE_DEFAULT’
 1329 |             FEATURE_DEFAULT(CanvasUsesAcceleratedDrawing),
      |             ^~~~~~~~~~~~~~~
```

--- a/Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml
+++ b/Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml
@@ -1665,6 +1665,7 @@ CanvasUsesAcceleratedDrawing:
   type: bool
   status: embedder
   humanReadableName: "Canvas uses accelerated drawing"
+  condition: USE(CA) || USE(SKIA)
   defaultValue:
     WebKitLegacy:
       "PLATFORM(IOS_FAMILY) && !PLATFORM(IOS_FAMILY_SIMULATOR)": true

--- a/Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp
+++ b/Source/WebKit/UIProcess/API/glib/WebKitSettings.cpp
@@ -1307,7 +1307,7 @@ static void webkit_settings_class_init(WebKitSettingsClass* klass)
             "enable-accelerated-2d-canvas",
             _("Enable accelerated 2D canvas"),
             _("Whether to enable accelerated 2D canvas"),
-            FEATURE_DEFAULT(CanvasUsesAcceleratedDrawing),
+            FALSE,
             static_cast<GParamFlags>(readWriteConstructParamFlags | G_PARAM_DEPRECATED));
 #endif
 
@@ -1326,7 +1326,11 @@ static void webkit_settings_class_init(WebKitSettingsClass* klass)
             "enable-2d-canvas-acceleration",
             _("Enable 2D canvas acceleration"),
             _("Whether to enable 2D canvas acceleration"),
+#if USE(SKIA)
             FEATURE_DEFAULT(CanvasUsesAcceleratedDrawing),
+#else
+            FALSE,
+#endif
             readWriteConstructParamFlags);
 
     /**

